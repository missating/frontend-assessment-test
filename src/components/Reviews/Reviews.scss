.reviews-container {
  display: flex;
  justify-content: flex-start;

  &__content {
    margin: 10px
  }

  h3 {
    margin: 0;
    font-size: 0.7rem;
    color: var(--brand-color);
  }

  p {
    margin:0;
    font-size: 0.5rem;
    color: rgb(30, 29, 34);
  }

  span {
    margin-top: 2px;
    font-size: 0.5rem;
  }
}
