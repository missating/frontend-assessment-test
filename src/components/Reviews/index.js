// thired party libraries
import React, { memo } from 'react'
import PropTypes from 'prop-types';

// styles
import './Reviews.scss'

const Reviews = ({ name, comment, positive }) => (
  <article className="reviews-container">
    <div className="reviews-container__content">
      <h3>{name}</h3>
      <p>{comment}</p>
      <span>
        Positive:
        {' '}
        {String(positive)}
      </span>
    </div>
  </article>
)

export default memo(Reviews);

Reviews.propTypes = {
  name: PropTypes.string.isRequired,
  comment: PropTypes.string.isRequired,
  positive: PropTypes.bool.isRequired,
}
