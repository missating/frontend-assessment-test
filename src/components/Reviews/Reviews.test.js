// third-party libraries
import React from 'react';
import { shallow } from 'enzyme';

// components
import Reviews from './index'

describe('Reviews Component', () => {
  const initialProps = {
    name: 'Susan Martins',
    comment: 'Nice hotel',
    positive: true,
  }
  it('should be rendered properly', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const wrapper = shallow(<Reviews {...initialProps} />);
    expect(wrapper.find('div').length).toEqual(1);
    expect(wrapper.find('article').length).toEqual(1)
    expect(wrapper).toMatchSnapshot();
  });
})
