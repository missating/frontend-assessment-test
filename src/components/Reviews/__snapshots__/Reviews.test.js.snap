// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews Component should be rendered properly 1`] = `
<article
  className="reviews-container"
>
  <div
    className="reviews-container__content"
  >
    <h3>
      Susan Martins
    </h3>
    <p>
      Nice hotel
    </p>
    <span>
      Positive:
       
      true
    </span>
  </div>
</article>
`;
